### PR TITLE
feat: Events endpoint

### PR DESCRIPTION
**Pull request by aomi**
Fixes tsoa by hoisting the types from the library. Adds a `events` collection and corresponding endpoint.

### CODE REVIEW

File: functions/src/events/Event.controller.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { add } from 'typesaurus';
+import { EventsCollection } from '../db/collections';
+
+@Route('events')
+export class EventsController extends Controller {
+  /**
+   * Retrieves all the courses available. If query params are passed in, they will be used to filter results.
+   * @param subject
+   * @param code
+   */
```
Comment (keithradford) [inline functions/src/events/Event.controller.ts:11]: 👀

File: functions/src/events/Event.controller.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  @Post()
+  public async postEvent(@Query() id: string): Promise<{ success: boolean }> {
```
Comment (keithradford) [inline functions/src/events/Event.controller.ts:13]: is `id` the best name? Not sure what you had in mind for intended use but if we're passing something like `Contest page visit` or something perhaps just `event` is better

**Review by keithradford — DISMISSED** [review]

**Comment by keithradford** [comment]
oops didn't mean to approve

File: functions/src/events/Event.controller.ts (hunk 2 above)
Comment (aomi) [reply functions/src/events/Event.controller.ts:13]: `id` just maps to what the URLs query params are where `id` is actually the location. I'll change it .

File: functions/src/events/Event.model.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export type EventRequest = {
+  name: string;
+  [key: string]: any;
```
Comment (keithradford) [inline functions/src/events/Event.model.ts:3]: I guess `any` is to keep it flexible with what we pass in? Should prob just add an `eslint-ignore` even though it's just a warning

**Review by keithradford — APPROVED** [review]

**Pull request merged**